test/formats/base_32.cpp

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright (c) 2011-2018 libbitcoin developers (see AUTHORS)
+ *
+ * This file is part of libbitcoin.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <boost/test/unit_test.hpp>
+#include <bitcoin/bitcoin.hpp>
+
+using namespace bc;
+
+BOOST_AUTO_TEST_SUITE(base_32_tests)
+
+bool decode_test(const std::string& in)
+{
+    std::string hr_part;
+    data_chunk values;
+    bool valid = libbitcoin::decode_base32(in, hr_part, values);
+}
+
+BOOST_AUTO_TEST_CASE(base32_encode_test)
+{
+    BOOST_REQUIRE_EQUAL(libbitcoin::encode_base32("a", data_chunk()), std::string("a12uel5l"));
+    BOOST_REQUIRE_EQUAL(libbitcoin::encode_base32("an83characterlonghumanreadablepartthatcontainsthenumber1andtheexcludedcharactersbio", data_chunk()), 
+        std::string("an83characterlonghumanreadablepartthatcontainsthenumber1andtheexcludedcharactersbio1tt5tgs"));
+}
+
+BOOST_AUTO_TEST_CASE(base32_decode_test)
+{
+    // Valid
+    BOOST_REQUIRE(decode_test("a12uel5l"));
+    BOOST_REQUIRE(decode_test("A12UEL5L"));
+    BOOST_REQUIRE(decode_test("an83characterlonghumanreadablepartthatcontainsthenumber1andtheexcludedcharactersbio1tt5tgs"));
+    BOOST_REQUIRE(decode_test("abcdef1qpzry9x8gf2tvdw0s3jn54khce6mua7lmqqqxw"));
+    BOOST_REQUIRE(decode_test("11qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqc8247j"));
+    BOOST_REQUIRE(decode_test("split1checkupstagehandshakeupstreamerranterredcaperred2y9e3w"));
+    BOOST_REQUIRE(decode_test("?1ezyfcl"));
```

Comment:
I'd prefer if you'd `BOOST_REQUIRE_EQUAL` for the decoded value similiar to what you did in the tests for `encode_base32()` instead of just checking for validity.